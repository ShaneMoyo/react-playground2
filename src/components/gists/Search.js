import React, {useState} from 'react'; 
import api from '../../services/api';
import PropTypes from 'prop-types';

export default function Search({ setGists }) { 
    const [username, setUsername] = useState(''); 
    const [loading, setLoading] = useState(false);
    const [error, setError] = useState(false);

    const handleChange = value => { 
        console.log(value); 
        setUsername(value); 
    }

    const  handleSubmit = async event => { 
        event.preventDefault(); 
        setLoading(true);
        setError(false)
        try {
            const result = await api.getGistByUsername(username);
            //debugger
            setGists(result); 
            setLoading(false);
            console.log('submit', result);
        } catch (err) {
            console.log('err: ', err);
            setError(true)
            setLoading(false);
        }     
    }

    return (
        <form onSubmit={event => handleSubmit(event)}>
            <input type="text" name="search" onChange={({ target: { value }}) => handleChange(value)}/>
            <br/>
            <button type="submit">Search</button> 
            {   loading ? <p>Loading...</p> : null  }
            {   error ? <p>Request failed...</p> : null  }
        </form>
    )
}

Search.propTypes = {
    setGists: PropTypes.func
};