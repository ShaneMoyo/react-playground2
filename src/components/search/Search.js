import React, {useState} from 'react'; 

export default function Search() { 
    const [search, setSearch] = useState(''); 

    const handleChange = value => { 
        console.log(value); 
        setSearch(value); 
    }

    const handleSubmit = event => { 
        event.preventDefault(); 
        console.log('submit', search)
    }

    return (
        <form onSubmit={event => handleSubmit(event)}>
            <input type="text" name="search" onChange={({ target: { value }}) => handleChange(value)}/>
            <br/>
            <button type="submit">Search</button> 
        </form>
    )
}