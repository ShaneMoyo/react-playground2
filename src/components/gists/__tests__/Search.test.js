import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Search from '../Search'; 



describe('components', () => {
   
  it('should render AddImage component', () => {
    const wrapper = shallow(<Search/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

});