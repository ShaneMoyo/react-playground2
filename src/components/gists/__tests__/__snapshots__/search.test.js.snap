// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components should render AddImage component 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="search"
    onChange={[Function]}
    type="text"
  />
  <br />
  <button
    type="submit"
  >
    Search
  </button>
</form>
`;
